be/src/exec/vectorized/aggregate/agg_hash_map.h

```diff
@@ -601,39 +609,27 @@ struct AggHashMapWithSerializedKeyFixedSize {
             key_column->serialize_batch(buffer, slice_sizes, chunk_size, max_fixed_size);
         }
 
-        FixedSizeSliceKey key;
-
-        if (!has_null_column) {
-            size_t* hash_values = reinterpret_cast<size_t*>(agg_states->data());
-            static_assert(sizeof(AggDataPtr) == sizeof(size_t));
-            for (size_t i = 0; i < chunk_size; i++) {
-                memcpy(key.u.data, buffer + i * max_fixed_size, max_fixed_size);
-                size_t hashval = hash_map.hash_function()(key);
-                hash_values[i] = hashval;
-            }
-            size_t __prefetch_index = AGG_HASH_MAP_DEFAULT_PREFETCH_DIST;
-
+        if (has_null_column) {
             for (size_t i = 0; i < chunk_size; ++i) {
-                if (__prefetch_index < chunk_size) {
-                    hash_map.prefetch_hash(hash_values[__prefetch_index++]);
-                }
-                memcpy(key.u.data, buffer + i * max_fixed_size, max_fixed_size);
-                auto iter = hash_map.lazy_emplace_with_hash(key, hash_values[i], [&](const auto& ctor) {
-                    AggDataPtr pv = allocate_func();
-                    ctor(key, pv);
-                });
-                (*agg_states)[i] = iter->second;
+                caches[i].key.u.size = slice_sizes[i];
             }
-        } else {
-            for (size_t i = 0; i < chunk_size; ++i) {
-                memcpy(key.u.data, buffer + i * max_fixed_size, max_fixed_size);
```

Comment:
we don't need this memcpy any more?

Reply:
No, we don't need this memory anymore. Looks at this code

```
    struct CacheEntry {
        FixedSizeSliceKey key;
        size_t hashval;
    };

    static constexpr size_t max_fixed_size = sizeof(CacheEntry);

        uint8_t* buffer = reinterpret_cast<uint8_t*>(caches.data());
        if (has_null_column) {
            memset(buffer, 0x0, max_fixed_size * chunk_size);
        }

        for (const auto& key_column : key_columns) {
            key_column->serialize_batch(buffer, slice_sizes, chunk_size, max_fixed_size);
        }
```
Because `key_columns` are serialized to `caches`, so we don't need memory copy anymore, just use `caches[i].key` to refer it.

And `memcpy` there is actually a `move` CPU instruction, not a function call. `mempcy` there is actually 

```
template <typename T>
inline T unaligned_load(const void* p) {
    T res{};
    memcpy(&res, p, sizeof(res));
    return res;
}
```